### PR TITLE
refactor: Split Track.hpp into TrackProxy.hpp and TrackContainer.hpp

### PR DESCRIPTION
**Pull request by paulgessinger**

### CODE REVIEW

File: Core/include/Acts/EventData/Track.hpp [hunk 1]
```diff
@@ -8,794 +8,6 @@
...
-  }
-};
-
-template <typename T>
-using TrackAccessor = TrackAccessorBase<T, false>;
-template <typename T>
-using ConstTrackAccessor = TrackAccessorBase<T, true>;
-
-}  // namespace Acts
+#include "Acts/EventData/TrackContainer.hpp"
```
Comment (paulgessinger) [inline Core/include/Acts/EventData/Track.hpp:13]: Keeping this header around for now, but printing a warning.

Comment (andiwand) [reply Core/include/Acts/EventData/Track.hpp:13]: is this to avoid breaking changes? do we have a followup for the next version on this?

**Review by andiwand — APPROVED** [review]
looks good to me

**Comment by paulgessinger** [comment]
Yeah I wanted to keep the other header around for another version or so @andiwand.

File: Core/include/Acts/EventData/Track.hpp (hunk 1 above)
Comment (paulgessinger) [reply Core/include/Acts/EventData/Track.hpp:13]: Yeah I wanted to keep the other header around for another version or so @andiwand.

**Pull request merged**